docs/guides/using-with-flakes-devshell.md

```diff
@@ -0,0 +1,162 @@
+`devenv` is partly a CLI, but also a way to configure a development shell declaratively. This configuration can also be used without using the `devenv` CLI inside other Nix flakes.
+
+Some usecases for using devenv configuration inside flakes is for projects that want to define other Nix flake features, apart from the development shell.
+These include a Nix package for the project, NixOS and home-manager modules related to the project.
+Usually you want to use the same lock file for the development shell as well as the Nix package and others, so that everything is based on the same nixpkgs.
+
+A Nix flake includes the inputs from `devenv.yaml` as well as the devenv configuration that you'd usually find in `devenv.nix`. `flake.lock` is the lock file for Nix flakes, the equivalent to `devenv.lock`.
+
+## Getting started
+
+To quickly set a project up with Nix flakes with devenv configuration, you can make use of `nix flake init`, like:
+
+```console
+$ nix flake init --template github:cachix/devenv
+```
+
+This will create a `flake.nix` with devnenv configuration, as well as a `.envrc` direnv configuration.
+
+You can open the devenv shell using:
+
+```console
+$ nix develop
+```
+
+This will create a lock file, if it wasn't already, and open up a new shell that adheres to the devenv configuration stated in `flake.nix`.
```

Comment:
```suggestion
This will create a lock file and open up a new shell that adheres to the devenv configuration stated in `flake.nix`.
```

It's not that important to explain all those edge cases :)